src/test/integration/integration-test.py

```diff
@@ -43,40 +79,14 @@ def test_get_non_authority_onids(self, endpoint='/spendingauthority'):
             self.assertEqual(response.json()['data'], spending_schema)
 
     # Test case: GET /spendingauthority with bad request
-    def test_error_response(self, endpoint='/spendingauthority'):
-        for onid in self.test_cases['bad_request']:
-            params = {'onid': onid}
+    def test_bad_request_response(self, endpoint='/spendingauthority'):
+        bad_requests = ["", {}, []]
+        for request in bad_requests:
+            params = {'onid': request}
```

Comment:
I guess you misunderstood. What I was saying for one of the bad request examples is `params = {}` instead of `params: {'onid': {}}`.